lib/srv/forward/sshserver.go

```diff
@@ -862,30 +937,54 @@ func (s *Server) rejectChannel(chans <-chan ssh.NewChannel, errMessage string) {
 }
 
 func (s *Server) handleGlobalRequest(ctx context.Context, req *ssh.Request) {
-	// Version requests are internal Teleport requests, they should not be
-	// forwarded to the remote server.
-	if req.Type == teleport.VersionRequest {
+	switch req.Type {
+	case teleport.VersionRequest:
+		// Version requests are internal Teleport requests, they should not be
+		// forwarded to the remote server.
 		err := req.Reply(true, []byte(teleport.Version))
 		if err != nil {
 			s.log.Debugf("Failed to reply to version request: %v.", err)
 		}
 		return
-	}
-
-	ok, payload, err := s.remoteClient.SendRequest(ctx, req.Type, req.WantReply, req.Payload)
-	if err != nil {
-		s.log.Warnf("Failed to forward global request %v: %v", req.Type, err)
-		return
-	}
-
-	if req.WantReply {
-		err = req.Reply(ok, payload)
+	case teleport.TCPIPForwardRequest, teleport.CancelTCPIPForwardRequest:
+		// Forwarding requests need to be authorized first.
+		if err := s.checkTCPIPForwardRequest(ctx, req); err != nil {
+			s.log.WithError(err).Warnf("Failed to check tcpip forward request")
+			if err := req.Reply(false, nil); err != nil {
+				s.log.Warnf("Failed to reply to global request: %v: %v", req.Type, err)
+			}
+			return
+		}
+		fallthrough
+	default:
+		ok, payload, err := s.remoteClient.SendRequest(ctx, req.Type, req.WantReply, req.Payload)
```

Comment:
Why are we ok forwarding random global requests?

Reply:
I'm not sure what we'd gain from blocking any particular request type, other than the authz for port forwarding that's already handled. Is there some request(s) you're thinking of that we should block?

Reply:
`streamlocal-forward@openssh.com` will open an AF_UNIX, SOCK_STREAM socket on the server, for example

Reply:
Fair enough. I've updated the handler to only allow whitelisted requests.